src/cpp/shared/combat/creatures.cpp

```diff
@@ -63,32 +63,39 @@ SecondaryResourceNameTableEntry SecondaryResourceNameTable[] = {
 	{"gems", RESOURCE_GEMS}
 };
 
-
-char* ironfistAttributeNames[] = {STRIKE_AND_RETURN, PLASMA_BLAST, TELEPORTER, ASTRAL_DODGE, SHADOW_MARK, JUMPER};
-int ironfistAttributeTable[ELEMENTS_IN(ironfistAttributeNames)][MAX_CREATURES];
+std::vector<std::string> ironfistAttributeNames = { STRIKE_AND_RETURN, PLASMA_BLAST, TELEPORTER, ASTRAL_DODGE, SHADOW_MARK, JUMPER};
+std::vector<std::vector<int>> ironfistAttributeTable;
 
 void ResetCreatureAttributes() {
-	memset(ironfistAttributeTable, 0, sizeof(ironfistAttributeTable));
+  ironfistAttributeTable.clear();
+  for(int j = 0; j < (int)ironfistAttributeNames.size(); j++) {
+    std::vector<int> tmp;
+    for(int i = 0; i < MAX_CREATURES; i++) {
+      tmp.push_back(0);
+    }    
+    ironfistAttributeTable.push_back(tmp);
+  }
 }
 
-void GrantCreatureAttribute(int id, const char* name) {
-	for(int i = 0; i < ELEMENTS_IN(ironfistAttributeNames); i++) {
-		if(0 == strcmp(ironfistAttributeNames[i], name)) {
+void GrantCreatureAttribute(int id, std::string name) {
+	for(int i = 0; i < (int)ironfistAttributeNames.size(); i++) {
+		if(ironfistAttributeNames[i] == name) {
 			ironfistAttributeTable[i][id] = 1;
 			return;
 		}
 	}
 
-	EarlyShutdown((char*)name, "Attempted to grant attribute that does not exist. Check creatures.xml .");
+	EarlyShutdown(&name[0], "Attempted to grant attribute that does not exist. Check creatures.xml .");
```

Comment:
This seems unsafe. Can you just use name.c_str()?

Reply:
c_str() returns const char* which is not char* that EarlyShutdown needs
https://stackoverflow.com/questions/347949/how-to-convert-a-stdstring-to-const-char-or-char